### DISCUSSION TITLE
#Beginner's Guide

### DISCUSSION THREAD

**Discussion by RajivRanasingha** [DuajDiaz/Travel-Website#23] [Q&A] [answered]
What is the difference between String, StringBuilder, and StringBuffer in Java?

**Comment by Bunny77K9** [comment, marked as answer]
The main difference is that String is immutable but both StringBuilder and StringBuffer are mutable. Also, StringBuilder is not synchronized like StringBuffer and that's why faster and should be used for temporary String manipulation.

**Reply by DuajDiaz** [reply]
Thank for your answer!
